app/lua/lua.c

```diff
@@ -22,498 +19,284 @@
 #include "lauxlib.h"
 #include "lualib.h"
 #include "legc.h"
-#include "lflash.h"
 #include "os_type.h"
 
+extern int debug_errorfb (lua_State *L);
+extern int pipe_create(lua_State *L);
+extern int pipe_read(lua_State *L);
+extern int pipe_unread(lua_State *L);
+
+#ifndef LUA_INIT_STRING
+#define LUA_INIT_STRING "@init.lua"
+#endif
+
+static int MLref = LUA_NOREF;
 lua_State *globalL = NULL;
 
-static lua_Load gLoad;
-static const char *progname = LUA_PROGNAME;
+static int pmain (lua_State *L);
+static int dojob (lua_State *L);
 
-static void l_message (const char *pname, const char *msg) {
-#if defined(LUA_USE_STDIO)
-  if (pname) c_fprintf(c_stderr, "%s: ", pname);
-  c_fprintf(c_stderr, "%s\n", msg);
-  c_fflush(c_stderr);
-#else
-  if (pname) luai_writestringerror("%s: ", pname);
+static void l_message (const char *msg) {
   luai_writestringerror("%s\n", msg);
-#endif
 }
 
-
 static int report (lua_State *L, int status) {
   if (status && !lua_isnil(L, -1)) {
     const char *msg = lua_tostring(L, -1);
     if (msg == NULL) msg = "(error object is not a string)";
-    l_message(progname, msg);
+    l_message(msg);
     lua_pop(L, 1);
   }
   return status;
 }
 
+static void l_print(lua_State *L, int n) {
+  lua_getglobal(L, "print");
+  lua_insert(L, -n-1);
+  if (lua_pcall(L, n, 0, 0) != 0)
+    l_message(lua_pushfstring(L, "error calling " LUA_QL("print") " (%s)",
+                              lua_tostring(L, -1)));
```

Comment:
I think this leaves the original error message on the stack here? Is that intentional?

Reply:
Not sure, but I immediately do a settop on return so the stack does balance -- I checked this one.

Reply:
Ah, all good then!